### PR TITLE
Option to hide mouse on inactivity

### LINKED ISSUE #4687: [Feature Request] Hide mouse cursor
An option that hides the mouse cursor would be greatly appreciated.

It could either be timed; so when the mouse cursor hasn't move in ~3 seconds - it hides.
Or it could be input based; so when you've pressed a button on the keyboard/gamepad - it hides.

This would be good in situations where you're playing a game that doesn't (or barely) requires touch input.
Having the cursor on the screen is very distracting.

### PR DESCRIPTION
**Pull request by vitor-k**
Add the option to hide the mouse cursor when a defined amount of time (currently ~~3s~~ 2.5s)
Adresses/closes #4687.

~~There are a few things that could be implemented with this, but I'm not sure if they're necessary or desirable.~~

- ~~Changing how long the mouse inactivity should be before hiding~~
- ~~Counting the mouse click as activity~~
- ~~Hotkey to show/hide the mouse~~
- ~~The ability to change the option while emulation is running~~

Mouse clicks now count as activity and the setting can be changed while emulation is running.

After reading the comments and some deliberation, I decided to not implement in this PR neither hotkeys nor changing the mouse timeout. Both are rather niche features and could be easily implemented by someone passionate about it, if they decide the current amount of control over the mouse hiding is not enough.

UI change:
hide-mouse

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5094)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -98,6 +98,8 @@ __declspec(dllexport) unsigned long NvOptimusEnablement = 0x00000001;
 }
 #endif
 
+const int default_mouse_timeout = 3000;
```
Comment (zhaowenlan1779) [inline src/citra_qt/main.cpp]:
```suggestion
static constexpr int default_mouse_timeout = 3000;
```

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -956,6 +964,12 @@ void GMainWindow::BootGame(const QString& filename) {
     }
     status_bar_update_timer.start(2000);
 
+    if (UISettings::values.hide_mouse) {
+        mouse_hide_timer.start(default_mouse_timeout);
```
Comment (zhaowenlan1779) [inline src/citra_qt/main.cpp]:
```suggestion
        mouse_hide_timer.start();
```

**Comment by adityaruplaha** [comment]
Can you please attach a screenshot as this is a UI change?

I haven't gone through the code yet, but since general suggestions based on my experience with apps that hide the mouse upon inactivity:

1. I think the inactivity time should be adjustable. Something like:
[x] Hide mouse pointer after: ______

2. Mouse clicks should be treated as activity. Most apps do this, so why not keep it intuitive for the user?

3. I don't think this requires a hotkey, as simply moving the mouse would make the pointer visible.

4. I wouldn't see that as a vital, but being able to change the setting during emulation would be a nice-to-have.

**Comment by BreadFish64** [comment]
> Changing how long the mouse inactivity should be before hiding
If it doesn't add complexity, sure
> Counting the mouse click as activity
Definitely
> Hotkey to show/hide the mouse
Not necessary
> The ability to change the option while emulation is running
That would be ideal

**Comment by jroweboy** [comment]
> Changing how long the mouse inactivity should be before hiding

I oppose adding an option for this. Just pick a good amount of time and leave it be. Someone somewhere will complain but its such an obscure setting that no one will ever want to maintain.

**Comment by Dragios** [comment]
A bit glitchy sometimes with this option enabled. Cursor disappear sometimes when clicking on menu despite with activity
image

**Comment by Dragios** [comment]
Looks good now. Can safely merge this.

**Pull request merged**